src/Compilers/CSharp/Portable/Symbols/Synthesized/SynthesizedImplementationMethod.cs

```diff
@@ -240,6 +240,11 @@ internal sealed override bool IsMetadataNewSlot(bool ignoreInterfaceImplementati
             return true;
         }
 
+        public override bool AreLocalsZeroed
+        {
+            get { return _implementingType.AreLocalsZeroed; }
```

Comment:
>get { return _implementingType.AreLocalsZeroed; } [](start = 12, length = 49)

It feels like prior implementations of AreLocalsZeroed that were not throwing should be changed to delegate to the type, unless there is an attribute on the method, or it is a lambda or a local function. Then we probably wouldn't even need to make any changes in this class.